pkg/appgw/certificates.go

```diff
@@ -132,10 +123,11 @@ func (c *appGwConfigBuilder) newHostToSecretMap(ingress *v1beta1.Ingress) map[st
 	return hostToSecretMap
 }
 
-func newCert(secretID secretIdentifier, cert *string) n.ApplicationGatewaySslCertificate {
+func (c *appGwConfigBuilder) newCert(secretID secretIdentifier, cert *string) n.ApplicationGatewaySslCertificate {
 	return n.ApplicationGatewaySslCertificate{
 		Etag: to.StringPtr("*"),
 		Name: to.StringPtr(secretID.secretFullName()),
+		ID:   to.StringPtr(c.appGwIdentifier.sslCertificateID(secretID.secretFullName())),
```

Comment:
It'd be nice if `c.appGwIdentifier.sslCertificateID` (and the rest of the `Identifier` methods) returned ptrs to strings already - would remove the need to use `to.StringPtr` here